khal/settings/utils.py

```diff
@@ -173,8 +173,8 @@ def config_checks(config):
             config['calendars'].pop(calendar)
     for vdir in sorted(vdirs):
         calendar = {'path': vdir,
-                    'color': get_color_from_vdir(vdir),
-                    'type': get_vdir_type(vdir),
+                    'color': None if testrun else get_color_from_vdir(vdir),
```

Comment:
I think it would result in nicer code if you add two extra parameters to `config_checks`: `get_color_from_vdir=get_color_from_vdir, get_vdir_type=get_vdir_type`. Then you can pass your custom variant of those functions in the test.